packages/phone-number-privacy/combiner/src/signing/get-threshold-signature.ts

```diff
@@ -195,19 +203,24 @@ async function handleSuccessResponse(
   const partialSig = { url: serviceUrl, signature: signResponse.signature }
   logger.info({ signer: serviceUrl }, 'Add signature')
   const signatureAdditionStart = Date.now()
-  await blsCryptoClient.addSignature(partialSig, blindedQueryPhoneNumber, logger)
+  await blsCryptoClient.addSignature(partialSig)
   logger.info(
     {
       signer: serviceUrl,
-      hasSufficientSignatures: blsCryptoClient.hasSufficientVerifiedSignatures(),
+      hasSufficientSignatures: blsCryptoClient.hasSufficientSignatures(),
       additionLatency: Date.now() - signatureAdditionStart,
     },
     'Added signature'
   )
   // Send response immediately once we cross threshold
-  if (blsCryptoClient.hasSufficientVerifiedSignatures()) {
-    // Close outstanding requests
-    controller.abort()
+  if (blsCryptoClient.hasSufficientSignatures()) {
+    try {
+      await blsCryptoClient.combinePartialBlindedSignatures(blindedQueryPhoneNumber, logger)
```

Comment:
I think we can also get rid of the if statement here and just use the try/catch

Reply:
We might also want to add a comment explaining why we need to try to combine the signatures early here

Reply:
> I think we can also get rid of the if statement here and just use the try/catch 
Given that most calls are expected to fail, it'll be nice to leave this check here as it helps self-document the expected behavior. It's an unexpected pattern to rely upon exceptions to be thrown when things are behaving as expected and might be misunderstood if someone were glancing at this code.

Reply:
Okay yea that makes sense 👍